packages/react-server/src/ReactFizzServer.js

```diff
@@ -816,10 +816,18 @@ function renderSuspenseBoundary(
   props: Object,
 ): void {
   if (someTask.replay !== null) {
-    throw new Error(
-      'Did not expect to see a Suspense boundary in this slot. ' +
-        "The tree doesn't match so React will fallback to client rendering.",
-    );
+    // If we're replaying through this pass, it means we're replaying through
+    // an already completed Suspense boundary. It's too late to do anything about it
+    // so we can just render through it.
```

Comment:
Not directly related to this PR but surfaced by the test cases.

Completed Suspense boundaries can be parents that we replay through but since they're already complete they don't add any features. It's just the same as if it's in the shell. All of it has already flushed.